edb/edgeql/ast.py

```diff
@@ -318,6 +318,7 @@ def from_python(cls, s: bytes) -> BytesConstant:
 
 class Parameter(Expr):
     name: str
+    optional: bool = True
```

Comment:
Are query arguments actually optional by default, e.g. would `fetchone('SELECT <int64>$0 +1', None)` work? Is that a good default?

```python
>>> c.fetchone('select <int64>$0 + 1', None)
Traceback (most recent call last):
  File "edgedb/protocol/protocol.pyx", line 409, in edgedb.protocol.protocol.SansIOProtocol._optimistic_execute
  File "edgedb/protocol/protocol.pyx", line 1089, in edgedb.protocol.protocol.SansIOProtocol.parse_data_messages
  File "edgedb/protocol/./codecs/./scalar.pyx", line 31, in edgedb.protocol.protocol.ScalarCodec.decode
  File "edgedb/pgproto/./codecs/int.pyx", line 118, in edgedb.pgproto.pgproto.int8_decode
  File "edgedb/pgproto/./frb.pxd", line 28, in edgedb.pgproto.pgproto.frb_read
  File "edgedb/pgproto/./frb.pyx", line 10, in edgedb.pgproto.pgproto.frb_check
AssertionError: insufficient data in buffer: requested 8 remaining 0

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/yury/dev/venvs/edgedb/lib/python3.8/site-packages/edgedb/blocking_con.py", line 107, in fetchone
    return self._protocol.sync_execute_anonymous(
  File "edgedb/protocol/blocking_proto.pyx", line 91, in edgedb.protocol.blocking_proto.BlockingIOProtocol.sync_execute_anonymous
  File "edgedb/protocol/blocking_proto.pyx", line 77, in edgedb.protocol.blocking_proto.BlockingIOProtocol._iter_coroutine
  File "edgedb/protocol/protocol.pyx", line 555, in execute_anonymous
  File "edgedb/protocol/protocol.pyx", line 445, in _optimistic_execute
edgedb.errors.ClientError: unable to decode data to Python objects
```

🤯

Reply:
> Is that a good default?

I'm actually thinking we should make query arguments required by default.

Reply:
I always assumed they are, so this is news to me. ;(